src/replication.c

```diff
@@ -3738,7 +3763,7 @@ void syncWithPrimary(connection *conn) {
     server.repl_transfer_read = 0;
     server.repl_transfer_last_fsync_off = 0;
     server.repl_transfer_lastio = server.unixtime;
-    return;
+    goto ok;
 
 no_response_error: /* Handle receiveSynchronousResponse() error when primary has no reply */
```

Comment:
These `goto` labels (`no_response_error`, `error`, `write_error`, and `ok`) have really gone too far now. Let's consolidate them into a single `syncWithPrimaryHandleError` function. I am hopeful that we can remove all the `goto`s in this function.